src/main/java/gift/service/WishService.java

```diff
@@ -24,36 +25,51 @@ public WishService(WishRepository wishRepository, ProductService productService)
 
     @Transactional
     public void save(MemberRequestDto memberRequestDto, WishRequestDto wishRequestDto){
-        ProductResponseDto productDto = productService.findByName(wishRequestDto.getProductName());
-        Product product = new Product(productDto.getId(), productDto.getName(), productDto.getPrice(), productDto.getImageUrl());
+        ProductResponseDto productResponseDto = productService.findByName(wishRequestDto.getProductName());
 
-        wishRepository.save(new Wish(memberRequestDto.toEntity(),product,wishRequestDto.getQuantity()));
+        Product product = productResponseDto.toEntity();
+        Member member = memberRequestDto.toEntity();
+
+        Wish newWish = new Wish(member,product,wishRequestDto.getQuantity());
+
+        product.addWish(newWish);
+        member.addWish(newWish);
```

Comment:
35,36 Line 의 코드는 비즈니스 로직과는 무관한 것 같아요.
35,36 Line 의 연관관계 편의 메서드를 Wish 를 생성하는 로직에 넣으면 어떨까요?
그러면 서비스단 로직이 더 깔끔해 질것 같아요!

```suggestion
        Wish newWish = Wish.of(member,product,wishRequestDto.getQuantity());
```

Reply:
Wish를 생성하는 로직에 추가하도록 수정하겠습니다!